stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -243,19 +253,49 @@ def register_loader_type(loader_type: type, provider_factory: Callable[[types.Mo
 def register_namespace_handler(importer_type: type, namespace_handler: _NSHandlerType) -> None: ...
 
 class IResourceProvider(IMetadataProvider): ...
-class NullProvider: ...
-class EggProvider(NullProvider): ...
+
+class NullProvider:
+    egg_name: str | None
+    egg_info: str | None
+    loader: types._LoaderProtocol | None
+    module_path: str | None
+
+    def __init__(self, module) -> None: ...
+    def get_resource_filename(self, manager, resource_name) -> str: ...
+    def get_resource_stream(self, manager, resource_name) -> BytesIO: ...
+    def get_resource_string(self, manager, resource_name): ...
+    def has_resource(self, resource_name) -> bool: ...
+    def has_metadata(self, name: str) -> bool | None: ...
+    def get_metadata(self, name: str) -> str: ...
+    def get_metadata_lines(self, name: str) -> Generator[str, None, None]: ...
+    def resource_isdir(self, resource_name) -> bool: ...
+    def metadata_isdir(self, name: str) -> bool: ...
+    def resource_listdir(self, resource_name) -> list[str]: ...
+    def metadata_listdir(self, name: str) -> list[str]: ...
+    def run_script(self, script_name: str, namespace: dict[str, Any]) -> None: ...
+
+class EggProvider(NullProvider):
+    egg_root: str
+
 class DefaultProvider(EggProvider): ...
 
 class PathMetadata(DefaultProvider, IResourceProvider):
+    egg_info: str
+    module_path: str
     def __init__(self, path: str, egg_info: str) -> None: ...
 
-class ZipProvider(EggProvider): ...
+class ZipProvider(EggProvider):
+    eagers: list[str] | None
+    zip_pre: str
 
 class EggMetadata(ZipProvider, IResourceProvider):
+    loader: types._LoaderProtocol
```

Comment:
This comes from the `zipimporter` argument to `__init__` so the types should match.